galadriel/tools/web3/market_data/dextools.py

```diff
@@ -0,0 +1,639 @@
+"""DEXTools API tools for market data using the dextools-python library."""
+
+from abc import ABC
+from enum import Enum
+import json
+import os
+import time
+from typing import Optional
+
+from dextools_python import DextoolsAPIV2
+
+from galadriel.tools import Tool
+from galadriel.logging_utils import get_agent_logger
+
+logger = get_agent_logger()
+
+# API configuration
+DEXTOOLS_API_KEY = os.getenv("DEXTOOLS_API_KEY")
+
+
+class DexToolsPlan(str, Enum):
+    """DEXTools API plans."""
+
+    FREE = "free"
+    TRIAL = "trial"
+    STANDARD = "standard"
+    ADVANCED = "advanced"
+    PRO = "pro"
+    PARTNER = "partner"
+
+    def __str__(self) -> str:
+        return self.value
+
+
+class DexToolsBase(Tool, ABC):
+    """Base class for DEXTools API tools."""
+
+    def __init__(self, plan: DexToolsPlan = DexToolsPlan.TRIAL):
+        super().__init__()
+        if not DEXTOOLS_API_KEY:
+            raise ValueError("DEXTOOLS_API_KEY environment variable is not set")
+        self.client = DextoolsAPIV2(DEXTOOLS_API_KEY, plan=str(plan))
+        self.plan = plan
+
+
+class GetBlockchainInfoTool(DexToolsBase):
+    """Tool for fetching blockchain information from DEXTools."""
+
+    name = "get_dextools_blockchain"
+    description = "Get blockchain information from DEXTools"
+    inputs = {
+        "chain": {
+            "type": "string",
+            "description": "The blockchain identifier (e.g., 'ether', 'bsc', 'polygon')",
```

Comment:
Shouldn't it be `ethereum`?

Reply:
it's ether defined by themselves unfortunately